python/lsst/utils/timer.py

```diff
@@ -94,9 +94,12 @@ def _find_outside_stacklevel() -> int:
     keyword parameter ``stacklevel``.
     """
     stacklevel = 1  # the default for `Logger.log`
-    stack = traceback.extract_stack()
-    for i, s in enumerate(reversed(stack)):
-        if "lsst/utils" not in s.filename:
+    stack = inspect.stack()
+    for i, s in enumerate(stack):
+        module = inspect.getmodule(s.frame)
+        if module is None:
+            continue
+        if not module.__name__.startswith("lsst.utils"):
```

Comment:
With inspect and frames you need to be very careful about creating local references to frames, as you can create reference cycles, so the fewer local variables you need the better, and cleaning up variables that contain frames before exiting a function is a good idea. This may be overly defensive for how often this function gets used, but it is good practice, especially if someone uses it as a code reference later on.
```suggestion
    for i, s in enumerate(inspect.stack()):
        module = inspect.getmodule(s.frame)
        if module is None:
            continue
        if not module.__name__.statswith("lsst.utils"):
            stacklevel = i
            break
    del s
    return stacklevel
```

Reply:
This seems highly unpythonic but I'll make the change.